src/components/Categorical.tsx

```diff
@@ -6,82 +6,160 @@ import {
+  const showStandardizedTerm =
+    standardizedVariable &&
+    standardizedVariable?.identifier !== getAssessmentToolConfig().identifier;
+
+  const getTermOptions = () => {
+    if (standardizedVariable?.identifier === 'nb:Diagnosis') {
+      return diagnosisTerms;
+    }
+    if (standardizedVariable?.identifier === 'nb:Sex') {
+      return sexTerms;
+    }
+    return [];
+  };
+
+  const termOptions = getTermOptions();
 
   return (
     <TableContainer
-      data-cy={`${columnId}-categorical`}
+      id={`${columnID}-table-container`}
       component={Paper}
       elevation={3}
-      className="h-full shadow-lg"
+      className="h-full shadow-lg overflow-auto"
+      style={{ maxHeight: '500px' }}
+      data-cy={`${columnID}-categorical`}
     >
-      <Table className="min-w-[768px]">
-        <TableHead data-cy={`${columnId}-categorical-table-head`}>
+      <Table stickyHeader className="min-w-[768px]">
+        <TableHead data-cy={`${columnID}-categorical-table-head`}>
           <TableRow className="bg-blue-50">
             <TableCell align="left" sx={{ fontWeight: 'bold', color: 'primary.main' }}>
               Value
             </TableCell>
             <TableCell align="left" sx={{ fontWeight: 'bold', color: 'primary.main' }}>
               Description
             </TableCell>
+            {showStandardizedTerm && (
+              <TableCell align="left" sx={{ fontWeight: 'bold', color: 'primary.main' }}>
+                Standardized Term
+              </TableCell>
+            )}
+            {standardizedVariable && (
+              <TableCell align="left" sx={{ fontWeight: 'bold', color: 'primary.main' }}>
+                Missing Value
+              </TableCell>
+            )}
           </TableRow>
         </TableHead>
         <TableBody>
-          {slicedValues.map((value) => (
-            <TableRow key={value} data-cy={`${columnId}-${value}`}>
+          {uniqueValues.map((value) => (
+            <TableRow key={`${columnID}-${value}`} data-cy={`${columnID}-${value}`}>
               <TableCell align="left">{value}</TableCell>
               <TableCell align="left">
                 <DescriptionEditor
-                  columnID={columnId}
+                  columnID={columnID}
                   levelValue={value}
                   description={levels[value]?.description || ''}
                   onDescriptionChange={(id, description) => {
                     onUpdateDescription(id, value, description || '');
                   }}
                 />
               </TableCell>
+              {showStandardizedTerm && (
+                <TableCell align="left">
+                  <Autocomplete
+                    options={termOptions}
+                    getOptionLabel={(option) => option.label}
+                    value={
+                      termOptions.find((opt) => opt.identifier === levels[value]?.termURL) || null
```

Comment:
this is honestly a bit hard to grasp. You're trying to fetch a label for a term if the term was already selected. That seems complex enough to lift to the state as a getter. Otherwise I have to have a pretty detailed mental model of what `levels` looks like and what `termOptions` looks like, and how they relate to each other. And I'd rather focus on this component.